elastic/elastic-agent
# Fix install/enroll command not  failing when the daemon restart fails

For the agent to be enrolled it needs to restart after the enrol process is completed, in order to pickup the new config and "connect" to fleet-server.

This change makes the enrol command to fail if it cannot restart the agent after enrolling on fleet. A `--skip-daemon-reload` CLI flag is also added to the `enroll` command because the `container` command calls `enroll`, however there is no running daemon to restart. Skipping the daemon reload on this case allows the `container` command to succeed.

This PR brings back the changes introduced by https://github.com/elastic/elastic-agent/pull/3554 that were reverted due to the `container` command not working and fixes the issues with the `container` command.

## What does this PR do?

This change makes the enroll command to fail if it cannot restart the agent after enrolling on fleet

## Why is it important?

For the agent to be enrolled it needs to restart after the enrol process is completed, in order to pickup the new config and "connect" to fleet-server.

## Checklist

~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~

~~## Author's Checklist~~


## How to test this PR locally
Export the follwing environment variables
```
export FLEET_ENROLL=1
export FLEET_URL=https://testing-this-PR.elastic-cloud.com:443
export FLEET_ENROLLMENT_TOKEN=my-secret-token
```
Then execute the `container` command: 
```
./elastic-agent container -e -v -d "*"
```

The Elastic-Agent should enrol in Fleet and start working.

## Related issues
- Closes https://github.com/elastic/elastic-agent/issues/3664
- Brings back: https://github.com/elastic/elastic-agent/pull/3554

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
